lib/src/cmaf/ham/manifestParser.ts

```diff
@@ -7,8 +20,52 @@ async function readHLS(manifestUrl: string): Promise<string> {
     return response.text();
 }
 
-
-export async function m3u8toHam(url: string) {
+export async function m3u8toHam(url: string): Promise<Presentation> {
+    const hls: string = await readHLS(url);
+    const parsedM3u8 = parseM3u8(hls);
+    const playlists: PlayList[] = parsedM3u8.playlists;
+    const mediaGroupsAudio = parsedM3u8.mediaGroups?.AUDIO;
+    let audioSwitchingSets: SwitchingSet[] = [];
+    let audioTracks : AudioTrack[] = [];
+    let selectionSets: SelectionSet[] = [];
     
+    for (let audio in mediaGroupsAudio) {
+        for (let attributes in mediaGroupsAudio[audio]) {
+            let language = mediaGroupsAudio[audio][attributes].language;
+            audioSwitchingSets.push(new SwitchingSet(audio, AUDIO_TYPE, '', 0, language,[]));
+            audioTracks.push(new AudioTrack(audio, AUDIO_TYPE, '', 0, language, 0, 0, 0,[]));
+        }
+    }
+
+    await Promise.all(playlists.map(async (playlist: any) => {
+        let manifestUrl = url.split("/").slice(0, -1).join("/") + "/" + playlist.uri;
+        let hlsManifest = await readHLS(manifestUrl);
+        let parsedHlsManifest = parseM3u8(hlsManifest);
+        let selectionSetDuration: number = 0;
+        let switchingSets: SwitchingSet[] = [];
+        let tracks: Track[]= [];
+        let audioId = playlist.attributes.AUDIO;
+
+        if (audioId) {
+            let audioSearched = audioSwitchingSets.find((audio) => audio.id === audioId);
+            let audioInSwitchingSet = switchingSets.find((audio) => audio.id === audioId);
+            if (audioSearched && !audioInSwitchingSet) switchingSets.push(audioSearched);
+        }
+
+        await Promise.all(parsedHlsManifest?.segments?.map(async (segment:any) => {
+            let {LANGUAGE, CODECS, BANDWIDTH } = playlist.attributes;
+            let {duration,uri} = segment;
+            let resolution = {width: playlist.attributes.RESOLUTION.width, height: playlist.attributes.RESOLUTION.height};
+            let { length, offset } = segment.byterange;
+            selectionSetDuration += duration;
+            let segments = [new Segment(duration ,uri,`${length}@${offset}`)];

```

Comment:
If the value is not reassigned, const should be used instead of let

Reply:
Resolved